ls1intum/Artemis
# Show related submissions of automatic feedback block in  Text Exercises

### Checklist

### Motivation and Context


Fixes #3524

### Description

Through this feature tutors can access and view the origin of automatically generated feedback on text submissions. The related submission is shown in a modal, where the re-used feedback is highlighted.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Find a TextExercise that has a an Automatic Feedback suggestion and click on its Find Origin Button next to the Automatic Label. Some use cases with Automatic Suggestions: 
    - [1 Automatic Feedback in 1 sentence submission](https://artemistest.ase.in.tum.de/course-management/482/text-exercises/7729/submissions/437965/assessment),
    - [1 Automatic Feedback in multiple sentence submission](https://artemistest.ase.in.tum.de/course-management/482/text-exercises/9578/submissions/439112/assessment), 
    - [1 Automatic Feedback with multiple origin feedbacks](https://artemistest.ase.in.tum.de/course-management/482/text-exercises/9579/submissions/439123/assessment),
    - [2 Automatic Feedbacks with Multiple sentences](https://artemistest.ase.in.tum.de/course-management/482/text-exercises/9579/submissions/439124/assessment)
4. Click on the Search icon next to the Automatic Label.
5. The modal should show. Check the screen recording below

### Test Coverage




### Screenshots

https://user-images.githubusercontent.com/18335489/124347972-8fb71200-dbe7-11eb-9fd2-094494b128ba.mp4

![image](https://user-images.githubusercontent.com/18335489/124347988-b6754880-dbe7-11eb-9a69-335fcf52ad35.png)

![image](https://user-images.githubusercontent.com/18335489/124349128-f3443e00-dbed-11eb-9b26-99b20e5f171c.png)
